### ISSUE TITLE
Audiveris hangs processing this sheet

### ISSUE THREAD

**Issue by kjancke** [Audiveris/audiveris#277]
I get a hang processing this sheet. Audiveris always runs into a timeout during header processing, I even increased the timeout to 5000s and the heap memory to 8G. CPU utilization stays at 100%.
![Prelude_0003](https://user-images.githubusercontent.com/19283886/56942377-a4d11b80-6b1a-11e9-90f4-6bded0ad1e92.jpg)

**Comment by hbitteur** [comment]
I found two kinds of problems with this example.

One is due to image quality. Even though the binarization step generally does a good job, here too many pixels are still considered as foreground, resulting in a lot of tiny glyphs. 
An enormous amount of CPU is spent in the headers step, when looking for potential time signature of first staff. The image below shows the rectangular region of interest which is being explored for a time sig, it contains 36 glyphs, resulting in 2**36 possible combinations to check just for this single location... 
The symbols step has protection about this potential explosion, the headers step has not. So this needs to be fixed.
![image](https://user-images.githubusercontent.com/22053634/56949105-4c0d7d00-6b32-11e9-8fb0-2d48e08191de.png)

The second problem is the number of cue notes. Systems 2 and 3, and beginning of system 4, are made of cue notes only. There are so many of them that the scale step has been fooled, considering that the typical beam height is 7 pixels (while the "standard" beam height is 12 in fact).
Moreover, there are no measure separations in systems 2 and 3, this will result is a very loooong measure.
And finally, the staves of systems 2 and 3, though linked by a brace in left margin, exhibit no joining barline except the system starting barline. Their staves are thus considered as independent parts, grouped by brace, but not a single part.

In short, first problem can easily be fixed, but the second cannot.

**Comment by hbitteur** [comment]
Quick update.
The high number of glyph parts in Time signature building resulted from the fact that all tiny glyphs were considered. For Clef, Key and Symbol building, a minimum weight is applied on glyph part. For some unknown reason, the same filtering for Time building had been commented out. I could not rewind far enough to discover when/why these code lines had been commented out, more than 5 years at least.
I will commit the fix.

Then, I had to manually inject beam height (using Sheet parameters) so that BEAMS step could run OK.

Allowing processing of small black notes (using Book parameters) was needed to get the cue heads recognized. This went OK.

Unfortunately, cue beams are not yet full-fledge citizens in current Audiveris. They are handled only in the vicinity of small heads, meant for grace notes. So, long cue beams of say several centimeters are not handled correctly.

**Comment by kjancke** [comment]
@Herve: thanks a lot for the quick update, i didn't expect such a quick response. I'll try the workarounds on the weekend. thanks again!

**Comment by kjancke** [comment]
I pulled the latest code and can confirm that it doesn't hang anymore, thanks!
I can also understand that the small notes are a problem, but I'd still want to give it a try - could you let me know what exact parameters you have set to which values?
Finally - although the processing finishes, there are lots of unrecognized notes:
![image](https://user-images.githubusercontent.com/19283886/57175446-7cb02800-6e4c-11e9-8e88-ec3e1cfebe8b.png)
Anything I can do about this?

**Comment by hbitteur** [comment]
You won't go far at all, because system 2, 3 and perhaps 4 can't be processed correctly. The reason is that long cue beams are not yet supported.

Just for your information:
1. Run until the SCALE step. At this point in time the typical beam height has been (wrongly) set to 7.
2. In pull-down menu "Sheet", select "Set Sheet Parameters...". Select the line "Beam thickness" by clicking on the left box. Then replace the value 7 by value 12. Click on "OK" to commit your action.
3. In pull-down menu "Book", select "Set Book Parameters...". Select the line "Support for small black note heads" by clicking on the left box. Then enable this option by clicking on the right box. Click on button "OK" to commit your action. This tells OMR engine to try to recognize cue heads (and small beams but you have only long beams)
4. Then continue processing...
